llvm/lib/Target/RISCV/RISCVTargetTransformInfo.cpp

```diff
@@ -475,10 +481,9 @@ InstructionCost RISCVTTIImpl::getShuffleCost(TTI::ShuffleKind Kind,
 
```

Comment:
Off topic for the actual change, but I think we can improve the code sequence above.

One idea would be to use a scalar multiply to do the broadcast since we know the value will either be 0 or (int16)-1.
Another would be to use a vrgather.vi in place of the extract/splat sequence.  
Another would be to extract the full16 bit value, do the extract and splat in scalar, and insert a full 16 bit value back.

Not sure if you care about this case or not, I don't much.

Reply:
Understood! Let's consider revisiting it when it becomes more relevant!